src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -304,6 +304,291 @@ public static void ReadPrimitiveList()
             Assert.Equal(2, i[1]);
         }
 
+        [Fact]
+        public static void ReadIEnumerableTOfIEnumerableT()
+        {
+            IEnumerable<IEnumerable<int>> result = JsonSerializer.Parse<IEnumerable<IEnumerable<int>>>(Encoding.UTF8.GetBytes(@"[[1,2],[3,4]]"));
```

Comment:
Glad to see nested arrays here - do we also have a check for nested arrays as an `IEnumerable` etc property on a class?

Reply:
I'll add these tests for following additional work we discussed about to parse nested arrays, list of lists, etc. as members of class objects (i.e. not just as root elements).

FYI I also added checks for the supported collection types as members of class objects.